core/src/main/java/io/aiven/klaw/helpers/db/rdbms/SelectDataJdbc.java

```diff
@@ -977,21 +977,22 @@ public List<ActivityLog> selectActivityLog(
 
   public List<Team> selectTeamsOfUsers(String username, int tenantId) {
     log.debug("selectTeamsOfUsers {}", username);
-    List<Team> allTeams = selectAllTeams(tenantId);
+    Map<Integer, Team> allTeams =
+        selectAllTeams(tenantId).stream()
+            .collect(Collectors.toMap(Team::getTeamId, Function.identity()));
 
     List<Team> teamList = new ArrayList<>();
-    List<UserInfo> userInfoList = Lists.newArrayList(userInfoRepo.findAllByTenantId(tenantId));
+    List<UserInfo> userInfoList =
+        Lists.newArrayList(userInfoRepo.findAllByTenantIdAndUsername(tenantId, username));
     Integer teamId;
```

Comment:
This all looks good but the username is the Id for the table, so we should be able to return just a single optional entry here instead of a list and just return a single optional value, I know the difference is marginal with that but I think it would make it clearer as currently it looks like we could potentially have multiple users with the same name when the UserInfo.java Pojo is annotated to prevent that.

Reply:
do i understand correctly that there couldn't be several records for the same `username` within this table?
E.g. even for different `tenantId`?

Reply:
Yeah, I checked out the UserInfo Pojo, only the username is annotated as the primary key for the table so each username is unique even across tenants. This is something we could improve potentially by changing the key to be username & tenantId but that we could do in a different PR.
https://github.com/Aiven-Open/klaw/blob/main/core/src/main/java/io/aiven/klaw/dao/UserInfo.java